File: demo-apps/TradecoreDemoObjectiveC/TradecoreDemoObjectiveC/TestCasesManager.m
//
//  TestCasesManager.m
//  TradecoreDemoObjectiveC
//
//  Created by Olena Stepaniuk on 15.05.2025.
//

#import "TestCasesManager.h"
#import "AdFormat.h"

#import "HTMLAdViewController.h"
#import "InstreamVideoIMAViewController.h"

@implementation TestCasesManager

+ (NSArray<TestCase *> *)allCases {
    return @[
        [
            [TestCase alloc]
            initWithTitle:@"HTML Ad View (Prebid + Amazon + GAM))"
            adFormats:@[@(AdFormatBanner)]
            configurationClosure:^UIViewController * _Nonnull{
                return [[HTMLAdViewController alloc] init];
            }
        ],
        
        [
            [TestCase alloc]
            initWithTitle:@"Instream Video (Prebid + GIMA)"
            adFormats:@[@(AdFormatInstreamVideo)]
            configurationClosure:^UIViewController * _Nonnull{
                return [InstreamVideoIMAViewController new];
            }
        ],
    ];
}

@end

File: demo-apps/TradecoreDemoAppSwift/TradecoreDemoAppSwift/TestCasesManager.swift
//
//  TestCasesManager.swift
//  TradecoreDemoAppSwift
//
//  Created by Olena Stepaniuk on 14.05.2025.
//

import UIKit

struct TestCase {
    let title: String
    let adFormats: Set<AdFormat>
    let configurationClosure: () -> UIViewController
}

struct TestCasesManager {
    
    static let allCases: [TestCase] = [
        
        TestCase(
            title: "HTML Ad View (Prebid + Amazon + GAM)",
            adFormats: [.banner],
            configurationClosure: {
                HTMLAdViewController()
            }
        ),
        
        TestCase(
            title: "Instream Video (Prebid + GIMA)",
            adFormats: [.instreamVideo],
            configurationClosure: {
                InstreamVideoIMAViewController()
            }
        ),
    ]
}
